src/main/java/com/google/cloud/teleport/templates/PubsubToAvro.java

```diff
@@ -123,9 +123,9 @@
 
     @Description("The maximum number of output shards produced when writing.")
     @Default.Integer(1)
-    Integer getNumShards();
+    ValueProvider<Integer> getNumShards();
```

Comment:
This will not work unfortunately as AvroIO.Write.withNumShards(int) does not support a ValueProvider as an input. The same goes for some of the other parameters below.